src/plugins/intel_npu/src/compiler/src/zero_compiler_in_driver.cpp

```diff
@@ -278,7 +277,7 @@ std::string LevelZeroCompilerInDriver<TableExtension>::serializeIOInfo(const std
 
         for (const std::shared_ptr<ov::op::v0::Parameter>& parameter : parameters) {
             const ov::element::Type& precision = parameter->get_element_type();
-            const size_t rank = parameter->get_shape().size();
+            const size_t rank = parameter->get_partial_shape().size();
```

Comment:
rank can also be dynamic

Reply:
@ArtemySkrebkov, do we need a throw for such cases? I'm pretty sure that we don't support this.

Reply:
Yep, we do not support dynamic rank.

Added a throw for this case. 

Previously, I thought to rely on `PartialShape::size` to throw (there is an assert for dynamic rank), but having a more precise error message sounds good to me too.